nerfstudio/process_data/hloc_utils.py

```diff
@@ -0,0 +1,54 @@
+from pathlib import Path
+from typing_extensions import Literal
+
+from nerfstudio.process_data.process_data_utils import CameraModel
+from hloc import extract_features, match_features, reconstruction, visualization, pairs_from_retrieval, pairs_from_retrieval, pairs_from_exhaustive
+
+from hloc.triangulation import (import_features, import_matches)
+from hloc.utils.io import get_keypoints, get_matches
+
+import pycolmap
```

Comment:
We don't have this as a dependency

Reply:
Is it possible to add these dependencies?
or should I make it as a separate script that can be called if the user installed the additional dependencies manually?

Reply:
hloc isn't provided as a pypi package which unfortunately means that we can't include it in our pypi package. We could add something like this - https://github.com/nerfstudio-project/nerfstudio/blob/main/nerfstudio/utils/install_checks.py

Reply:
Also is `pycolmap` necessary? We include many colmap utils here - https://github.com/nerfstudio-project/nerfstudio/blob/main/nerfstudio/process_data/colmap_utils.py

Reply:
pycolmap is a dependency of hloc, and the pycolmap enums are used to configure the reconstruction